### DISCUSSION TITLE
the font not shown in Program!

### DISCUSSION THREAD

**Discussion by mahi200300** [dotnet/csharplang#8403] [Q&A]
I wrote a Windows Form program and used a special font in the program! In the name of "FredokaOne.ttf", I have installed the font and the font is displayed well in the program, but it was not displayed in my friend's system, which does not have the font installed.

program in my pc:
![program in my pc](https://github.com/user-attachments/assets/f0686715-028a-4aba-a4e0-8d8d19f91913)

program in Other pc:
![program in Other pc](https://github.com/user-attachments/assets/afe9a6bc-2fb6-4617-963e-4737fd31fa4d)

I added that font to my project's resources, but it still didn't change :
```
public Form1()
{
InitializeComponent();
LoadCustomFont();
}
private void LoadCustomFont()
{
PrivateFontCollection privateFonts = new PrivateFontCollection();
// Load the font from resources
byte[] fontData = Properties.Resources.FredokaOne_Regular; // Assuming the font file is named FredokaOne.ttf
IntPtr fontPtr = Marshal.AllocCoTaskMem(fontData.Length);
Marshal.Copy(fontData, 0, fontPtr, fontData.Length);
privateFonts.AddMemoryFont(fontPtr, fontData.Length);
Marshal.FreeCoTaskMem(fontPtr);
// Apply the font to a control
button1.Font = new Font(privateFonts.Families[0], 12,FontStyle.Regular);
copy_button.Font = new Font(privateFonts.Families[0], 12, FontStyle.Regular);
label1.Font = new Font(privateFonts.Families[0], 12, FontStyle.Regular);
label2.Font = new Font(privateFonts.Families[0], 12, FontStyle.Regular);
}
```

**Comment by colejohnson66** [comment]
This is not a language question. It belongs more on a site like StackOverflow.

Also, there's a use-after-free bug in your program. You can't `FreeCoTaskMem` before you stop using the font. The allocated memory *must* remain allocated until the constructed `Font` instances are destroyed.

**Reply by mahi200300** [reply]
tnx but in C# we did not manually manage memory!

**Comment by jaredpar** [comment]
Closing as this is more of a WinForms / general programming question while this discussion forum is more for C# language questions.

**Reply by jnm2** [reply]
https://github.com/dotnet/winforms/discussions may be a good place to ask!

**Reply by mahi200300** [reply]
my program is coded in C# too :/

**Reply by CyrusNajmabadi** [reply]
This is for questions/design of the language itself. Not external libraries. Thanks.
